pkg/azure/client_test.go

```diff
@@ -0,0 +1,41 @@
+// +build unittest
+
+package azure
+
+import (
+	"errors"
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+type FakeSender struct {
+	statusCode int
+}
+
+func (fs FakeSender) Do(request *http.Request) (response *http.Response, err error) {
+	response = &http.Response{
+		StatusCode: fs.statusCode,
+	}
+	if fs.statusCode != 200 {
+		err = errors.New("Error while making a GET for the gateway")
+	}
+	return response, err
+}
+
+var _ = DescribeTable("Az Application Gateway failures using authorizer", func(statusCodeArg int) {
+	var azClient = NewAzClient("", "", "", "", "")
+	var fakeSender = FakeSender{
+		statusCode: statusCodeArg,
+	}
+	azClient.SetDuration(3)
```

Comment:
reduce the duration to something in milliseconds

Reply:
changed code to take in string input where its easier to specify duration